File: Counting Elements/FrogRiverOne/FrogRiverOne.js
function solution(X, A) {
    const positions = [];
    let earliestTime = -1;
    let noJumps = 0;
    for(let i = 0; i < A.length; i++) {
        if(positions[A[i]]) {
            continue;
        }
        positions[A[i]] = true;
        noJumps++;
        if(noJumps === X) {
            earliestTime = i;
            break;
        }
    }
    return earliestTime;
}

const A = [1,3,1,4,2,3,5,4];
const X = 5;

console.log(solution(X, A));

File: Counting Elements/FrogRiverOne/FrogRiverOne.py
def solution(X, A):
    B = [None] * X
    for k,v in enumerate(A):
        if B[v-1] == None:
             B[v-1] = k
    
    
    for v in B:
        if v == None:
            return -1
    B.sort()   
    
    return B[X-1]